src/update-feeds-loops/get-updatable-feeds.ts

```diff
@@ -19,122 +19,152 @@ export interface UpdatableBeacon {
-        if (modifiedHeartbeatInterval < 0n) {
-          logger.warn('Resulting heartbeat interval is negative. Setting it to 0.', {
-            dapiName: decodedDapiName,
-            dataFeedId,
-            heartbeatInterval,
-            heartbeatIntervalModifier,
-          });
-        }
-        const adjustedHeartbeatInterval = modifiedHeartbeatInterval < 0n ? 0n : modifiedHeartbeatInterval;
-
-        // We need to make sure that the update transaction actually changes the on-chain value. There are two cases:
-        // 1. Data feed is a beacon - We need to make sure the off-chain data updates the feed. This requires timestamp
-        //    to change. See:
-        //    https://github.com/api3dao/airnode-protocol-v1/blob/65a77cdc23dc5434e143357a506327b9f0ccb7ef/contracts/api3-server-v1/DataFeedServer.sol#L120
-        // 2. Data feed is a beacon set - We need make sure that the beacon set will change. The contract requires the
-        //    beacon set value or timestamp to change. See:
-        //    https://github.com/api3dao/airnode-protocol-v1/blob/65a77cdc23dc5434e143357a506327b9f0ccb7ef/contracts/api3-server-v1/DataFeedServer.sol#L54
-        const isBeaconSet = beaconValues.length > 1;
-        // Note, that the beacon set value/timestamp is computed as median, so single beacon update may not result in a beacon set update.
-        if (isBeaconSet && newDataFeedValue === dataFeedValue && newDataFeedTimestamp === dataFeedTimestamp) {
-          return false;
-        }
-        if (!isBeaconSet && newDataFeedTimestamp === dataFeedTimestamp) return false;
-
-        return logger.runWithContext(
-          {
-            dapiName: decodedDapiName,
-            dataFeedId,
-          },
-          () =>
-            checkUpdateCondition(
-              dataFeedValue,
-              dataFeedTimestamp,
-              newDataFeedValue,
-              newDataFeedTimestamp,
-              adjustedHeartbeatInterval,
-              adjustedDeviationThresholdCoefficient,
-              deviationReference
-            )
-        );
-      })
-      // Compute the updateable beacons.
-      .map(({ dataFeedInfo, aggregatedBeaconsWithData }) => ({
+        },
+        () =>
+          checkUpdateCondition(
+            dataFeedValue,
+            dataFeedTimestamp,
+            newDataFeedValue,
+            newDataFeedTimestamp,
+            adjustedHeartbeatInterval,
+            adjustedDeviationThresholdCoefficient,
+            deviationReference
+          )
+      );
+
+    if (dataFeedNeedsUpdate) {
+      acc.push({
         dataFeedInfo,
         updatableBeacons: aggregatedBeaconsWithData
           .filter(({ isUpdatable }) => isUpdatable)
           .map(({ beaconId, signedData }) => ({
             beaconId,
             signedData: signedData!,
           })),
-      }))
-  );
+        shouldUpdateBeaconSet: isBeaconSet,
+      });
+    } else if (isBeaconSet && asyncBeaconUpdatedeviationThresholdFactor) {
```

Comment:
You don't need to check `isBeaconSet` here afaik.

Reply:
I think I do in the case where: 
1. data feed is a beacon (isBeaconSet is false)
2. isValidBeaconUpdate is true
3. checkUpdateCondition() returns false
4. individualBeaconUpdateDeviationThresholdCoefficient is set in the config